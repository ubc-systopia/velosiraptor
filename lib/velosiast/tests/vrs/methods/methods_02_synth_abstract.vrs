// abstract method
abstract fn foo() -> bool
    requires true;

// synthesis method
synth fn foo() -> bool
    requires true;


synth fn foo() -> bool
    requires true;

extern fn foo() -> bool

extern fn foo() -> bool
    requires true;

