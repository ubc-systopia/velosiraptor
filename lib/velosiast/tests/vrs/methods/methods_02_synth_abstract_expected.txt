abstract fn foo() -> bool
  requires true;

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

synth fn foo() -> bool
  requires true;

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

synth fn foo() -> bool
  requires true;

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

extern fn foo() -> bool
  requires true;

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

extern fn foo() -> bool
  requires true;

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

